tests/scripts/set-release-ver.sh

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+FROM_TAG="$1"
+TO_TAG="$2"
+
+
+# This script is supposed to facilitate the preparation of a rook release by
+# changing example and documentation files to use the new release version instead the old one.
+#
+# It will usually be run on a release branch and the resulting changes will be the content of the
+# commit for the basis of the release.
+
+# The script needs two version tags:
+
+# FROM_TAG: the version tag of the previous/current release (e. g. v1.16.7)
+# TO_TAG: the version tag of the new release being prepared (e. g. v1.16.8)
+
+# The two tags should be passed to the script as the first two positional parameters in its invocation.
+
+# So an example invocation could be:
+# set-release-ver.sh v1.16.7 v1.16.8
+#
+# The script replaces all occurrences of $FROM_TAG with $TO_TAG in
+# the relevant affected files.
+
+
+
+
+FILES=(
+Documentation/Getting-Started/quickstart.md
+Documentation/Storage-Configuration/Monitoring/ceph-monitoring.md
+Documentation/Upgrade/rook-upgrade.md
+deploy/charts/rook-ceph/values.yaml
+deploy/examples/direct-mount.yaml
+deploy/examples/images.txt
+deploy/examples/operator-openshift.yaml
+deploy/examples/operator.yaml
+deploy/examples/osd-purge.yaml
+deploy/examples/toolbox-job.yaml
+deploy/examples/toolbox-operator-image.yaml
+)
+
+sed -i -e "s/$FROM_TAG/$TO_TAG/g" "${FILES[@]}"
```

Comment:
This command will be too general. If the tag is `master`, we will end up replacing too many references. Instead, let's use very specific `sed` commands to avoid changing too many places. For example:
```
"s/docker.io/rook/ceph:$FROM_TAG/docker.io/rook/ceph:$TO_TAG/g" 
"s/tag: $FROM_TAG/tag: $TO_TAG/g" 
```
And there are others, basically we should end up with the same results as we see in a previous PR such as #15737. It should be simple to run the script and inspect that we get the same results as that PR.

Reply:
@travisn  wrote:
> This command will be too general. If the tag is `master`, we will end up replacing too many references. Instead, let's use very specific `sed` commands to avoid changing too many places. For example:
> 
> ```
> "s/docker.io/rook/ceph:$FROM_TAG/docker.io/rook/ceph:$TO_TAG/g" 
> "s/tag: $FROM_TAG/tag: $TO_TAG/g" 
> ```

sounds good. I was trying to implement something along those lines.

But I think this might be too narrow: It might miss the documentation pieces.

I might end up using different sed rules for different sets of files. - wdyt?


> And there are others, basically we should end up with the same results as we see in a previous PR such as #15737. It should be simple to run the script and inspect that we get the same results as that PR.